packages/number-field/src/vaadin-number-field.js

```diff
@@ -187,13 +224,19 @@ export class NumberField extends InputFieldMixin(InputSlotMixin(ThemableMixin(El
     this._increaseValue();
   }
 
-  /** @protected */
+  /**
+   * @protected
+   * @override
+   */
   _createConstraintsObserver() {
     // NOTE: do not call "super" but instead override the method to add extra arguments
     this._createMethodObserver('_constraintsChanged(required, min, max, step)');
   }
 
-  /** @private */
+  /**
+   * @protected
+   * @override
+   */
   _constraintsChanged(required, min, max) {
```

Comment:
Can't see `step` in the list of arguments. That probably will prevent the validation in case `step` has been changed. Is it intended behavior?

Reply:
Added. Please note however, this does not affect the execution of the observer.
Only the arguments in the declaration of the observer matter, and these are listed in `static get constraints()`.

Reply:
Yeah, that's true. Although, my question was more about: should the input be re-validated once the user has changed the `step` attribute?

Reply:
> Should the input be re-validated once the user has changed the step attribute as well as it happens for min and max attributes?

Yes, probably. Let's not change it in this PR, I will create a separate issue instead.